src/api/server.rs

```diff
@@ -538,24 +518,21 @@ impl ServerClient {
         &self,
         request: &words::delete::Request,
     ) -> Result<words::delete::Response> {
-        match self
+        let resp = self
             .client
             .post(format!("{}/words/delete", self.api))
             .query(request)
             .send()
             .await
-        {
-            Ok(resp) => {
-                match resp.error_for_status_ref() {
-                    Ok(_) => {
-                        resp.json::<words::delete::Response>()
-                            .await
-                            .map_err(Error::ResponseDecode)
-                    },
-                    Err(_) => Err(Error::InvalidRequest(resp.text().await?)),
-                }
+            .map_err(Error::Reqwest)?;
+
+        match resp.error_for_status_ref() {
+            Ok(_) => {
+                resp.json::<words::delete::Response>()
+                    .await
+                    .map_err(Error::Reqwest)
             },
-            Err(e) => Err(Error::RequestEncode(e)),
+            Err(_) => Err(Error::InvalidRequest(resp.text().await?)),
```

Comment:
I think we should add a test for this.

E.g., sending a request that is too long. I don't know that is the limit of local servers, but the public API has a limit of 1500 characters, I think.

Reply:
Sure, I'll have a look

Reply:
Is that last commit there what you had in mind?

Reply:
Yes, it looks good @Rolv-Apneseth! Sorry, I didn't see the notification...

Could you fix the conflicts so I can run tests?

Reply:
No issue, ping me when you're back ;-)

Reply:
@jeertmans Did a rebase, hopefully it's all good now

Reply:
Perfect, thanks! We will fix CodSpell's warnings later (we need to ignore them)